### PR TITLE
Makefile: replace DEFINE to DEFINE_PREFIX

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
For more information see:
https://github.com/apache/incubator-nuttx/pull/7076

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: crypto/mbedtls/Make.defs [hunk 1]
```diff
@@ -26,7 +26,7 @@ CONFIGURED_APPS += $(APPDIR)/crypto/mbedtls
 CFLAGS += ${shell $(INCDIR) "$(CC)" $(APPDIR)/crypto/mbedtls/mbedtls/include}
 CXXFLAGS += ${shell $(INCDIR) "$(CC)" $(APPDIR)/crypto/mbedtls/mbedtls/include}
 
-CFLAGS += ${shell $(DEFINE) "$(CC)" MBEDTLS_CONFIG_FILE='"<crypto/mbedtls_config.h>"'}
-CXXFLAGS += ${shell $(DEFINE) "$(CC)" MBEDTLS_CONFIG_FILE='"<crypto/mbedtls_config.h>"'}
+CFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE='"<crypto/mbedtls_config.h>"'}
+CXXFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE='"<crypto/mbedtls_config.h>"'}
```
Comment (pkarashchenko) [inline crypto/mbedtls/Make.defs:30]:
```suggestion
CFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE='"<crypto/mbedtls_config.h>"'
CXXFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE='"<crypto/mbedtls_config.h>"'
```

File: crypto/mbedtls/Makefile [hunk 2]
```diff
@@ -36,7 +36,7 @@ MBEDTLS_UNPACKLIBDIR = $(MBEDTLS_UNPACKNAME)$(DELIM)library
 MBEDTLS_UNPACKPROGDIR = $(MBEDTLS_UNPACKNAME)$(DELIM)programs
 
 # This lets Mbed TLS better use some of the POSIX features we have
-CFLAGS += ${shell $(DEFINE) "$(CC)" __unix__}
+CFLAGS += ${DEFINE_PREFIX}__unix__}
```
Comment (pkarashchenko) [inline crypto/mbedtls/Makefile:39]:
```suggestion
CFLAGS += ${DEFINE_PREFIX}__unix__
```

File: system/argtable3/Makefile [hunk 3]
```diff
@@ -28,7 +28,7 @@ ARGTABLE3_SRCDIR   = $(ARGTABLE3_UNPACK)$(DELIM)src
 DEPPATH += --dep-path $(ARGTABLE3_SRCDIR)
 VPATH   += :$(ARGTABLE3_SRCDIR)
 
-CFLAGS += ${shell $(DEFINE) "$(CC)" ARG_REPLACE_GETOPT=0}
+CFLAGS += ${DEFINE_PREFIX}ARG_REPLACE_GETOPT=0}
```
Comment (pkarashchenko) [inline system/argtable3/Makefile:31]:
```suggestion
CFLAGS += ${DEFINE_PREFIX}ARG_REPLACE_GETOPT=0
```

Comment (Gary-Hobson) [reply system/argtable3/Makefile]: done~

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**